mesonbuild/mesonmain.py

```diff
@@ -54,6 +54,16 @@ def setup_vsenv():
         return
     if 'Visual Studio' in os.environ['PATH']:
         return
+    # VSINSTALL is set when running setvars from a Visual Studio installation
+    # Tested with Visual Studio 2012 and 2017
+    if 'VSINSTALLDIR' in os.environ:
+        return
+    # Check explicitly for cl when on Windows
+    # Windows XP, 2000 and 10 reports 'Windows_NT'. I assume that all other Windows versions in between do the same.
+    if (os.environ.get('OS', bat_placeholder) == 'Windows_NT'):
```

Comment:
There is no need to check for Windows (and not in this way) when there already is a "mesonlib.is_windows()" check in the start of the function. I will remove that in yet another commit